### PR TITLE
4817 react router upgrade

### PR DESCRIPTION
**Pull request by keithjgrant**
##### SUMMARY

Upgrades React & React Router to the latest versions

* Reworked tests to use `act()` where applicable
* Updated `mountWithContexts` to use the updated React Router context

Addresses #4669 #4817

### CODE REVIEW

**Comment by keithjgrant** [comment]
recheck

File: awx/ui_next/src/index.jsx [hunk 1]
```diff
@@ -56,31 +55,38 @@ export function main(render) {
...
-                <Route exact path="/" render={defaultRedirect} />
-                <Route
-                  render={() => (
+            <Switch>
+              {removeTrailingSlash}
+              <Route path="/login" render={defaultRedirect} />
+              <Route exact path="/" render={defaultRedirect} />
+              <Route
+                render={() => {
+                  if (!isAuthenticated(document.cookie)) {
```
Comment (marshmalien) [inline awx/ui_next/src/index.jsx:86]: 👍 Very cool. I like this method of conditionally routing/rendering within the render func.

File: awx/ui_next/src/screens/Template/shared/JobTemplateForm.test.jsx [hunk 2]
```diff
@@ -71,14 +72,17 @@ describe('<JobTemplateForm />', () => {
...
+    await act(async () => {
+      wrapper = mountWithContexts(
+        <JobTemplateForm
+          template={mockData}
+          handleSubmit={jest.fn()}
+          handleCancel={jest.fn()}
+        />
+      );
+    });
+    // await waitForElement(wrapper, 'Form', el => el.length === 0);
```
Comment (marshmalien) [inline awx/ui_next/src/screens/Template/shared/JobTemplateForm.test.jsx]: Should we remove this line?

**Review by marshmalien — APPROVED** [review]
👏 This looks great. Thanks for digging into the login and logout <Redirect> bugs. 

Keith and I chatted offline, thanks to the upgrade, we could update the skipped test in `JobDetail.test.jsx` by rolling it into this PR or a separate PR.

File: awx/ui_next/src/screens/Template/shared/JobTemplateForm.test.jsx (hunk 2 above)
Comment (keithjgrant) [reply awx/ui_next/src/screens/Template/shared/JobTemplateForm.test.jsx]: done

**Pull request merged**